### PR TITLE
seastar,crimson: pickup change to pin socket to fixed core

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@ronen-fr what do you mean by "what was not merged from Seastar tip"? i think https://github.com/ceph/seastar/commits/ceph-octopus-19.06.0-414-g77b23a18 is rebased on the HEAD of scylladb/seastar's master branch.

**Comment by ronen-fr** [comment]
@tchaikov what I meant is that I didn't notice an immediate way to find what commits in seastar are not picked up by this PR. Take a look at how github presents the change in the subcomponent for review: it appears as a list of commits, with no "context".
Anyway - it's not a problem with this PR. I'll try to find the easiest way to translate this data for myself for next time.

**Comment by ronen-fr** [comment]
And again - perf test was OK yesterday, failed today, after a change to comments formatting. The problem seems to be with stddev comparisons, which fail also on my laptop - for whatever version. It may be that we've set a too-low threshold there.

jenkins test crimson perf

**Comment by tchaikov** [comment]
@ronen-fr it's just a few clicks away: https://github.com/scylladb/seastar/compare/master...ceph:77b23a18613d2e7f09f749aa10061dd3ee3e6032

**Comment by ronen-fr** [comment]
@tchaikov ... that was pretty stupid of me...

**Comment by tchaikov** [comment]
```
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 11.3404/11.1645  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 2903.0/2858.0  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 122.981/179.862  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.00551021/0.005579  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 11.2766/11.1967  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 2886.0/2866.0  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 113.239/184.785  => accepted
08:14:14 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.00552523/0.00556286  => accepted
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 50.0103/51.3373  => accepted
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 12802.0/13142.0  => accepted
08:14:14 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 1179.74/298.603  => rejected
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.00124237/0.00120729  => accepted
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 50.1132/51.5494  => accepted
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 12828.0/13196.0  => accepted
08:14:14 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 1221.8/332.231  => rejected
08:14:14 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.00123966/0.0012038  => accepted
08:14:14 - WARNING  - cbt      - 2 tests failed out of 16
```

File: src/crimson/common/errorator.h [hunk 1]
```diff
@@ -701,6 +705,11 @@ struct errorator {
   future<T...> make_exception_future2(std::exception_ptr&& ex) noexcept {
     return future<T...>(exception_future_marker(), std::move(ex));
   }
+  template <typename... T>
+  static
+  future<T...> make_exception_future2(seastar::future_state_base&& state) noexcept {
+    return future<T...>(exception_future_marker(), std::move(state));
+  }
```
Comment (rzarzynski) [inline src/crimson/common/errorator.h]: The `errorator`-related changes look good.

**Pull request merged**